src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,101 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const INITIAL_FORM = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [initialForm, setInitialForm] = useState(INITIAL_FORM);
+  const [count, setCount] = useState(0);
+
+  const handleFormChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInitialForm((prevForm) => ({
+      ...prevForm,
+      [event.target.name]: event.target.value,
+    }));
+  };
+
+  const isAddButtonDisabled = !initialForm.title
+    || !initialForm.imgUrl
+    || !initialForm.imdbUrl
+    || !initialForm.imdbId;
+
+  const reset = () => {
+    setInitialForm(INITIAL_FORM);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onAdd({
+      title: initialForm.title,
+      description: initialForm.description,
+      imgUrl: initialForm.imgUrl,
+      imdbUrl: initialForm.imdbUrl,
+      imdbId: initialForm.imdbId,
+    });
```

Comment:
Here you can just pass your state(but remember about renaming)

```suggestion
    onAdd(initialForm);
```